rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,133 @@
+
+![Figure 1](../images/0000-multi-data-collection-data-processing-images/project-submission-in-project.png)
+
+
+## Application to analysis pipelines
+
+While this RFC proposes a general mechanism, it was developed in response to the needs of analysis to co-process data that are in multiple assay bundles.  This section uses co-processing of multiple sequencing run from a given library as an example of the use of data groups to identify data that should be processed together.  A key concept behind this proposal is that upstream submission should not be defining how downstream analysis groups inputs for processing.  The analysis pipelines need to be able to group input based on ways that may not be defined at submission time.  New analysis pipelines should be able to group inputs in arbitrary ways without requiring changes to the way data is packaged.
+
+### Identifying data to co-process
+Pipelines that require grouping data from multiple sequencing assays (co-processing) subscribe to *PROJECT_SUBMISSIONS* events instead of per-bundle events. Unlike with the assay bundle per pipeline run, this makes the pipeline framework responsible for collecting assay bundles into pipeline runs.  Pipelines that do not require co-processing continue to subscribe to assay bundle events and are unaffected.
+
+For full background and a detailed proposal on how these libraries are represented in the metadata see the RFC [Representing sequencing library preparations in the HCA DCP metadata standard](https://github.com/HumanCellAtlas/dcp-community/blob/68f0c42fc9bd55297be8c241663721271c90b694/rfcs/text/0010-rfc-library-preparation.md).
+
+The *data group* submission event does not define bundles to be co-processed, it indicates that data to be co-processed is contained within the *data group* and that that group fulfills version-completeness requirements. Co-processing will be initiated by pipelines based on a well-defined set of rules that is data modality-specific.  The partition of a submission *data group* into co-processing units may require multiple parameters. For instance, with 10X V2 3’ scRNA-seq is grouped by library, however, a submission may contain non-10X data modalities that need to be processed separately.
+
+An analysis pipeline event handler component will be created that is responsible for partitioning a *data group* into co-processing units, dispatching the pipelines, and tracking completion. This is the responsibility of Analysis.  When analysis results are submitted to ingest, this creates a secondary analysis *data group*.
```

Comment:
The idea is that these analyses still need to generate their own bundle notifications (as at present) but also need to generate a data group notification? This is definitely going to get us into some intricacies of how analysis will signal those groups, since we can't simply say that everything in a project is in the data group.

Reply:
My understanding is the DSS generates the events when actions are taken on a bundle.  So analysis doesn't directly signal. 

We could put all bundles in a single per project group and let the client sort it out by introspection.  However, this is less obvious and probably a lot of overhead.  We are updated the RFC to make this clearer.